slashing-protection/src/test/java/tech/pegasys/web3signer/slashingprotection/RegisteredValidatorsTest.java

```diff
@@ -0,0 +1,111 @@
+    final RegisteredValidators registeredValidators =
+        new RegisteredValidators(mockJdbi, validatorsDao, registeredValidatorsMap);
+    registeredValidatorsMap.put(PUBLIC_KEY1, 1);
+    registeredValidatorsMap.put(PUBLIC_KEY2, 2);
+
+    assertThat(registeredValidators.getValidatorIdForPublicKey(PUBLIC_KEY1)).hasValue(1);
+    assertThat(registeredValidators.getValidatorIdForPublicKey(PUBLIC_KEY2)).hasValue(2);
+  }
+
+  @Test
+  public void retrievesEmptyValidatorIdForUnregisteredValidator() {
+    final RegisteredValidators registeredValidators =
+        new RegisteredValidators(mockJdbi, validatorsDao, HashBiMap.create());
+    assertThat(registeredValidators.getValidatorIdForPublicKey(PUBLIC_KEY3)).isEmpty();
+  }
+
+  @Test
+  public void retrievesPublicKeyForRegisteredValidator() {
+    final BiMap<Bytes, Integer> registeredValidatorsMap = HashBiMap.create();
+    final RegisteredValidators registeredValidators =
+        new RegisteredValidators(mockJdbi, validatorsDao, registeredValidatorsMap);
+    registeredValidatorsMap.put(PUBLIC_KEY1, 1);
+    registeredValidatorsMap.put(PUBLIC_KEY2, 2);
+
+    assertThat(registeredValidators.getPublicKeyForValidatorId(1)).hasValue(PUBLIC_KEY1);
+    assertThat(registeredValidators.getPublicKeyForValidatorId(2)).hasValue(PUBLIC_KEY2);
+  }
+
+  @Test
+  public void retrievesEmptyPublicKeyForUnregisteredValidator() {
+    final RegisteredValidators registeredValidators =
+        new RegisteredValidators(mockJdbi, validatorsDao, HashBiMap.create());
+    assertThat(registeredValidators.getPublicKeyForValidatorId(1)).isEmpty();
+  }
+
+  @Test
+  public void retrievesAllValidatorIds() {
+    final BiMap<Bytes, Integer> registeredValidatorsMap = HashBiMap.create();
+    final RegisteredValidators registeredValidators =
+        new RegisteredValidators(mockJdbi, validatorsDao, registeredValidatorsMap);
+    registeredValidatorsMap.put(PUBLIC_KEY1, 1);
+    registeredValidatorsMap.put(PUBLIC_KEY2, 2);
+    registeredValidatorsMap.put(PUBLIC_KEY3, 3);
+
+    assertThat(registeredValidators.validatorIds()).isEqualTo(Set.of(1, 2, 3));
+  }
+
+  @Test
+  public void registersValidatorsThatAreNotAlreadyInDb(final Jdbi jdbi) {
+    final BiMap<Bytes, Integer> registeredValidatorsMap = HashBiMap.create();
+    final RegisteredValidators registeredValidators =
+        new RegisteredValidators(jdbi, validatorsDao, registeredValidatorsMap);
+
+    when(validatorsDao.registerValidators(any(), any())).thenCallRealMethod();
+
+    registeredValidators.registerValidators(List.of(PUBLIC_KEY1));
+    assertThat(registeredValidatorsMap).hasSize(1);
+
+    registeredValidators.registerValidators(List.of(PUBLIC_KEY1, PUBLIC_KEY2, PUBLIC_KEY3));
+    assertThat(registeredValidatorsMap).hasSize(3);
+    // because 'id' is a sequence, the values will be 1, 2, 3
+    assertThat(registeredValidatorsMap)
+        .isEqualTo(Map.of(PUBLIC_KEY1, 1, PUBLIC_KEY2, 2, PUBLIC_KEY3, 3));
+  }
+}
```

Comment:
missing tests for mustGetValidatorIdForPublicKey?

Reply:
Yes it is. Added some.